Fix Icon alignment issues for Web :CategorySearch & DataSearch, and Vue: DataSearch

mohdashraf010897 opened the pull request:
- **Issue Type**: Bug
- **Description**: Icon Alignment in the search input is not proper with absurd margins and padding, changing the style of the input field is distorting the icons' position. Also, the icons used are not from a consistent source. 


### _WEB_-**CategorySearch**, **DataSearch**   &&   _VUE_-**DataSearch**
1. Files Added: In the Styles folder, 2 files are added: 
    - src---> styles---> **IconGroup.js**: Wrapper for multiple icons, uses flexbox for content alignment and absolute positioning for self-alignment. 
    -  src---> styles---> **IconWrapper.js**: Wrapper container for icon images.
2. Files Changed: 
    - src---> styles---> **Input.js**: passed a prop that accounts for whether the voice-search icon is also present and adjusts the input field padding accordingly.
    -  src---> components---> search---> **CategorySearch.js**: Updated to use updated icon rendering implementation using IconWrapper and IconGroup.
    - src---> components---> search---> **DataSearch.js**: Replaced using InputIcon.js to IconGroup.js and IconWrapper.js
     - src---> components---> shared---> **CancelSvg.js**: Applied close icon svg from Material library.


Notion Link - https://www.notion.so/appbase/ReactiveSearch-Icon-alignment-Issues-bdd2ec34e3894715a5e1692b1c4782a3

Demo - 

[https://www.loom.com/share/9cfe9996041d4ba5a35b59b91a111d9c](https://www.loom.com/share/9cfe9996041d4ba5a35b59b91a111d9c)

mohdashraf010897 committed:
Fixed icon alignment issues for CategorySearch

mohdashraf010897 committed:
Fixed icon alignment issues for DataSearch : React

mohdashraf010897 committed:
Fixed icon alignment issues for DataSearch : React

mohdashraf010897 committed:
Fixed icon alignment issues for DataSearch : Vue

mohdashraf010897 committed:
Fixed icon alignment issues for RS: web - CategorySearch and DataSearch, and RS: Vue - DataSearch

bietkul commented on packages/web/src/styles/IconWrapper.js:
```diff
@@ -0,0 +1,22 @@
+import styled from '@emotion/styled';
+import { css } from '@emotion/core';
```
@mohdashraf010897 Do you not get eslint error? You're not using `css` anywhere. Check it at other places too.

mohdashraf010897 replied:
**React-RS** --- DataSearch Component demo
[https://www.loom.com/share/e81e0cce748f4d7986dcc61e21bc0faa](https://www.loom.com/share/e81e0cce748f4d7986dcc61e21bc0faa)

**React-RS** --- CategorySearch Component demo
[https://www.loom.com/share/5f43418f5ef94dc7a66fa4493e895f77](https://www.loom.com/share/5f43418f5ef94dc7a66fa4493e895f77)

bietkul commented on packages/web/src/components/shared/SearchSvg.js:
```diff
@@ -7,6 +7,7 @@ const SearchSvg = () => (
 		height="12"
 		xmlns="http://www.w3.org/2000/svg"
 		viewBox="0 0 15 15"
+		style={{transform:'scale(1.25)', position:'relative'}}
```
@mohdashraf010897 Avoid using `inline` styles everywhere. It would be hard to customize it.

mohdashraf010897 replied:
done
<img width="379" alt="Screenshot 2021-05-26 at 2 14 29 AM" src="https://user-images.githubusercontent.com/57627350/119565894-24984580-bdc8-11eb-89cf-ea948af7d3e5.png">

bietkul commented on packages/web/src/components/shared/CancelSvg.js:
```diff
@@ -2,22 +2,10 @@ import React from 'react';
-		c6.989,6.99,18.322,6.99,25.312,0c6.99-6.99,6.99-18.322,0-25.312L131.804,106.491z"
-		/>
-	</svg>
+
```
@mohdashraf010897 Why is this not formatted? Do you not have the `prettier` installed?

mohdashraf010897 replied:
done 
<img width="575" alt="Screenshot 2021-05-26 at 2 15 42 AM" src="https://user-images.githubusercontent.com/57627350/119566037-4d203f80-bdc8-11eb-8b82-e8cc62c75bf6.png">

bietkul commented on packages/web/src/components/search/addons/Mic.js:
```diff
@@ -152,14 +153,13 @@ class Mic extends React.Component {
 		}
 		return (
-			<MicIcon showIcon={showIcon} showClear={applyClearStyle} iconPosition={iconPosition}>
+			<IconWrapper>
```
@mohdashraf010897 Is there any reason to remove the props? How did you test?

mohdashraf010897 replied:
All these 3 props were used by the styled component wrapper to adjust the position of the icon in respect to the search-icon's position and the existence of a clear icon, which is now taken care of in the Input.js styled component and the applied IconGroup.js styled component making the props redundant and hence removed.

bietkul commented on packages/web/src/components/search/DataSearch.js:
```diff
@@ -702,38 +704,49 @@ class DataSearch extends Component {
-					/>
-				)}
-				<InputIcon
+					<div>
```
@mohdashraf010897 Formatting issues

bietkul commented on packages/web/examples/CategorySearch/src/index.js:
```diff
@@ -27,6 +27,9 @@ const Main = () => (
 					enableRecentSearches
 					enablePopularSuggestions
 					size={5}
+					showClear
```
@mohdashraf010897 You can remove these changes.

mohdashraf010897 replied:
done

bietkul commented on packages/vue/src/components/shared/SearchSvg.js:
```diff
@@ -6,6 +6,7 @@ export default () => (
 		width="10"
 		xmlns="http://www.w3.org/2000/svg"
 		viewBox="0 0 15 15"
+		style={{transform:'scale(1.5)', position:'relative'}}
```
@mohdashraf010897 Avoid `inline` style

bietkul commented on packages/vue/src/components/shared/CancelSvg.js:
```diff
@@ -1,17 +1,5 @@
-			}
-		/>
-	</svg>
+	<svg alt="Clear"
```
@mohdashraf010897 Formatting issues, `vue` folder has a different eslint config so you can open it in a new editor window and format it.

bietkul reviewed (changes requested):
@mohdashraf010897 PR looks good 👍 .
It would be great if you can share a demo to cover the following scenarios:
- Change the `iconPosition` prop to `left` and `right`
- Use `innerClass` prop to modify the icon style
- Try with custom search icon
- Share demos for `vue-RS` and `vue-searchbox` too.

bietkul commented:
@mohdashraf010897 Resolve the conflicts as well.

mohdashraf010897 committed:
Fixed linting errors

Linting erros in various files have been fixed, and inline styles have been avoided

mohdashraf010897 committed:
Fixed linting errors

Linting erros in various files have been fixed, and inline styles have been avoided

mohdashraf010897 committed:
Merge branch 'next' into fix/icon-alignment-issues

mohdashraf010897 committed:
Merge branch 'next' into fix/icon-alignment-issues

mohdashraf010897 committed:
Fixed padding and margin issues with default icons

bietkul commented on packages/vue/src/components/search/DataSearch.jsx:
```diff
@@ -691,35 +692,43 @@ const DataSearch = {
-						</InputIcon>
-					)}
-					{showVoiceSearch && (
+					<IconGroup groupPosition="right"
```
@mohdashraf010897 Formatting is off

bietkul commented on packages/vue/src/components/search/addons/Mic.jsx:
```diff
@@ -137,14 +138,14 @@ const Mic = {
 		}
 		return (
-			<MicIcon showIcon={showIcon} showClear={applyClearStyle} iconPosition={iconPosition}>
+			<IconWrapper>
```
@mohdashraf010897 Formatting issues

bietkul commented on packages/vue/src/styles/Input.js:
```diff
@@ -8,6 +8,7 @@ const alertBorder = ({ theme }) => css`
 const input = css`
 	width: 100%;
 	height: 42px;
```
Formatting issues

bietkul commented on packages/web/examples/DataSearch/src/index.css:
```diff
@@ -146,3 +146,7 @@ body {
 	background-size: cover;
 }
+
+.search-icon {
```
Remove it

bietkul commented on packages/web/src/components/search/CategorySearch.js:
```diff
@@ -795,37 +794,41 @@ class CategorySearch extends Component {
+					groupPosition="left"
+					positionType="absolute"
+				>{iconPosition === 'left' && <IconWrapper	onClick={this.handleSearchIconClick}>{this.renderIcon()}</IconWrapper>}
+				</IconGroup>
```
Formatting issues

bietkul commented on packages/web/src/components/search/DataSearch.js:
```diff
@@ -720,35 +725,34 @@ class DataSearch extends Component {
-				</InputIcon>
+				<IconGroup groupPosition="right" positionType="absolute">
+					{this.state.currentValue && showClear && (
+						<IconWrapper onClick={this.clearValue} showIcon={showIcon} isClearIcon>
```
Formatting issues

bietkul commented on packages/web/src/components/shared/CancelSvg.js:
```diff
@@ -3,20 +3,18 @@ import types from '@appbaseio/reactivecore/lib/utils/types';
+		height="20px"
+		viewBox="0 0 24 24"
+		width="20px"
+		fill="#000000"
```
@mohdashraf010897 Hardcoded `fill` value could break for existing users. This is how we're merging classes. Make that change everywhere.
![alt image](https://i.imgur.com/K7cPmBt.png)

mohdashraf010897 replied:
Couldn't clearly understand? Does this work? have removed hard-coded fill from the svg file itself.

<img width="675" alt="Screenshot 2021-05-26 at 12 18 04 PM" src="https://user-images.githubusercontent.com/57627350/119614928-78347e80-be1c-11eb-9ce1-9f137e19da98.png">

mohdashraf010897 committed:
Merge branch 'next' into fix/icon-alignment-issues

mohdashraf010897 committed:
Fixed Requested changes

mohdashraf010897 committed:
Resolved Conflicts
